pkg/dsl/definition/template.go

```diff
@@ -119,86 +112,145 @@ func (wd *workloadDef) Complete(ctx process.Context) error {
 				if err != nil {
 					return errors.WithMessagef(err, "parse WorkloadDefinition %s outputs(%s)", wd.name, fieldInfo.Name)
 				}
-				ctx.PutAssistants(process.Assistant{Ins: other, Type: AuxiliaryWorkload})
+				ctx.PutAuxiliaries(process.Auxiliary{Ins: other, Type: AuxiliaryWorkload, Name: fieldInfo.Name, IsOutputs: true})
 			}
 		}
 	}
 	return nil
 }
 
-// Output fetch the workload cr and set result to context
-func (wd *workloadDef) Output(ctx process.Context, client client.Client, name string) Template {
-	base, _ := ctx.Output()
+func (wd *workloadDef) getTemplateContext(ctx process.Context, cli client.Reader, ns string) (map[string]interface{}, error) {
+
+	var commonLabels = map[string]string{}
+	var root = map[string]interface{}{}
+	for k, v := range ctx.BaseContextLabels() {
+		root[k] = v
+		switch k {
+		case "appName":
+			commonLabels[oam.LabelAppName] = v
+		case "name":
+			commonLabels[oam.LabelAppComponent] = v
+		}
+	}
+
+	base, assists := ctx.Output()
 	componentWorkload, err := base.Unstructured()
 	if err != nil {
-		return wd
+		return nil, err
 	}
-	workloadCr, err := getObj(client, componentWorkload, name)
+	// workload main resource will have a unique label("app.oam.dev/resourceType"="WORKLOAD") in per component/app level
+	object, err := getResourceFromObj(componentWorkload, cli, ns, util.MergeMapOverrideWithDst(map[string]string{
```

Comment:
I don't fully understand this. The object is not emitted to the K8s yet, how can we get the object through the client?

Reply:
In this stage,  the resource has already ommited to K8s, just like health check